developer-docs-site/static/messageboard_demo/client/README.md

```diff
@@ -0,0 +1,187 @@
+...
+}
+```
+
+The participant can then claim this capability and then move it to their own account. Later on, the participant can use this resource stored in their account to modify the pinned_post
+
+```tsx
+let notice_cap = Offer::redeem<MessageChangeCapability>(
+            account, board);
+move_to(account, notice_cap);
+```
+
+Move also support emitting events and these events will be stored in event stream that can be queried through API request. For example, any one can send a message to the messageboard if they don’t want to modify the pinned post. The contract will emit an event and add the event to the board’s event stream. Thus anyone subscribe to the event stream of the board can see all the messages. 
+
+```tsx
+public(script) fun send_message_to(
+    account: signer, board_addr: address, message: vector<u8>
+) acquires MessageChangeEventHandle {
+    let event_handle = borrow_global_mut<MessageChangeEventHandle>(board_addr);
+    Event::emit_event<MessageChangeEvent>(
+        &mut event_handle.change_events,
+        MessageChangeEvent{
+            message,
+            participant: Signer::address_of(&account)
+        }
+    );
+}
+```
+
+### ACL-Based Messageboard
+
+[ACL-Based messageboard](https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/move-examples/messageboard/sources/ACLMessageBoard.move) is similar to the capability based messageboard. The main difference is it uses ACL to control the access, which is essentially a list of the addresses. 
+
+To add a new participant, the board creator adds the participant into the ACL list under his account. 
+
+```tsx
+public(script) fun add_participant(account: &signer, participant: address) acquires ACLBasedMB {
+    let board = borrow_global_mut<ACLBasedMB>(Signer::address_of(account));
+    ACL::add(&mut board.participants, participant);
+}
+```
+
+When a participant want to change the pinned message, the participant first need to borrow the board resource under creator’s account and check if the participant is in the ACL. Then it borrows mutable board to modify the pinned_post. 
+
+```tsx
+public(script) fun send_pinned_message(
+    account: &signer, board_addr: address, message: vector<u8>
+) acquires ACLBasedMB, MessageChangeEventHandle {
+    let board = borrow_global<ACLBasedMB>(board_addr);
+    assert!(ACL::contains(&board.participants, Signer::address_of(account)),EACCOUNT_NOT_IN_ACL);
+
+    let board = borrow_global_mut<ACLBasedMB>(board_addr);
+    board.pinned_post = message;
+```
+
+## Interacting With Messageboard
+
+### **Compile and Publish the Modules**
+
+The [following code](https://github.com/aptos-labs/aptos-core/blob/main/developer-docs-site/static/messageboard_demo/client/index.ts) basically runs a command to compile the messageboard contract with the account address of developer. 
+
+It then read the compiled modules in mv files in the module path and creates a txn to publish the modules under the developer’s account address.
```

Comment:
read → reads
mv → mv (Move)
does txn mean transaction? If yes, then better to say "transaction".

Reply:
updated